fcrepo-kernel-modeshape/src/main/java/org/fcrepo/kernel/modeshape/FedoraBinaryImpl.java

```diff
@@ -383,15 +384,33 @@ public RdfStream getFixity(final IdentifierConverter<Resource, FedoraResource> i
     }
 
     @Override
-    public Collection<URI> checkFixity(final IdentifierConverter<Resource, FedoraResource> idTranslator,
-                               final Collection<String> algorithms) throws UnsupportedAlgorithmException {
+    public Collection<URI> checkFixity( final IdentifierConverter<Resource, FedoraResource> idTranslator,
+                                        final Collection<String> algorithms)
+                                            throws UnsupportedAlgorithmException, UnsupportedAccessTypeException {
 
         fixityCheckCounter.inc();
 
         try (final Timer.Context context = timer.time()) {
 
-            LOGGER.debug("Checking resource: " + getPath());
-            return CacheEntryFactory.forProperty(getProperty(JCR_DATA)).checkFixity(algorithms);
+            final String mimeType = getMimeType();
+            if (mimeType.indexOf(CacheEntryFactory.MESSAGE_EXTERNAL_BODY) >= 0) {
+                final Map<String, String> params = CacheEntryFactory.parseExternalBody(mimeType);
+                final String accessType = params.get("access-type");
+                final String resourceLocation = params.get(accessType.toLowerCase());
+
+                if (accessType != null && accessType.equals("URL")) {
+                    LOGGER.debug("Checking external resource: " + resourceLocation);
+
+                    return CacheEntryFactory.forProperty(getProperty(HAS_MIME_TYPE)).checkFixity(algorithms);
+                } else {
+                    throw new UnsupportedAccessTypeException("Unsupported access-type " + accessType
```

Comment:
I would expect the following command (with JUNK access-type) to fail... but it succeeds:
```
curl -i -ufedoraAdmin:fedoraAdmin -XPUT -H"Content-Type: message/external-body;access-type=JUNK;url=\"https://blognumbers.files.wordpress.com/2010/09/6.jpg\"" localhost:8080/rest/image

```
I see it fails when requesting `Want-Digest`... but should also fail at the point of creation. That can be a new, follow-on ticket.
https://fcrepo.github.io/fcrepo-specification/#external-content

Reply:
Nevermind. I see this issue is addressed in: https://jira.duraspace.org/browse/FCREPO-2586